python/cudf/cudf/_lib/aggregation.pyx

```diff
@@ -253,6 +253,34 @@ cdef class Aggregation:
         ))
         return agg
 
+    # scan aggregations
```

Comment:
See my suggestion in groupby.pyx. I think we can just reuse the non-cumulative versions of these functions and remove all of these classmethods entirely since they're identical at the level of the creation of the C++ aggregation. This will map more closely to the C++ code since it retains a 1-1 mapping from cudf's `AggregationKind` to libcudf's `aggregation.Kind`.

Reply:
Right now, they are identical. When https://github.com/rapidsai/cudf/issues/7106 is implemented, they will not be identical. Hence separate functions.
Besides, `cumcount` is different.

One suggestion to avoid repetition could be to create alias.
```python
    cumsum = sum
    cummin = min
    cummax = max
```

Reply:
Yes I like that solution. I agree that this will change when the C++ side is modified to be stricter about providing the types available, but until then I think it's helpful to match the C++ as closely as possible as well as avoiding duplication here if we can.